qutebrowser/utils/utils.py

```diff
@@ -838,3 +838,20 @@ def mimetype_extension(mimetype: str) -> Optional[str]:
     if mimetype in overrides:
         return overrides[mimetype]
     return mimetypes.guess_extension(mimetype, strict=False)
+
+
+@contextlib.contextmanager
+def cleanup_file(filepath: str) -> Iterator[None]:
+    """Context that deletes a file upon exit or error.
+
+    Args:
+        filepath: The file path
+    """
+    try:
+        yield
+    finally:
+        try:
+            if os.path.exists(filepath):
```

Comment:
I still don't think this is needed because we handle `OSError` anyways:

```pycon
>>> try:
...     os.remove('/doesnotexist')
... except OSError as e:
...     print(f"Failed to delete file: {e}")
... 
Failed to delete file: [Errno 2] No such file or directory: '/doesnotexist'
```

Reply:
Removed if-statement